### DISCUSSION TITLE
covariance function to handle None

### DISCUSSION THREAD

**Discussion by santiromoli** [cornellius-gp/gpytorch#2219] [Ideas]
I think it will be very useful to many users to have a covarience function that can handle 'None' type. I work in chemogenomics and it is very common to use datasets with many 'None' type inputs, simply because that information does not exist.

**Comment by Balandat** [comment]
Handle how? What do you propose that covariance function should do with None values?

**Reply by santiromoli** [reply]
I guess I'd try two ways:
- a simple one, a cov that ignores 'None'
- an algorithm inside the cov that interpolates the 'None'

**Comment by gpleiss** [comment]
If `None` represents missing data, there's lots of ways that you could design a kernel to handle the missing data. The most straightforward idea however is to simply augment your training data by imputing all missing features (via interpolation, mean value, etc.). Then you wouldn't have any `None` values in your data and you could use any standard kernel.
